h2o-py/h2o/explanation/_explain.py

```diff
@@ -1226,6 +1226,33 @@ def pd_multi_plot(
             plt.savefig(fname=save_plot_path)
         return decorate_plot_result(figure=fig)
 
+def _handle_orig_values(orig_value, is_factor, tmp, encoded_col, plt, target, model, frame,
+                        index, column, color, percentile_string):
+    PDP_RESULT_FACTOR_NAN_MARKER = '.missing(NA)'
+    user_splits = dict()
+    if _isnan(orig_value):
+        if is_factor:
+            idx = np.where(tmp[encoded_col] == tmp.from_factor_to_num(encoded_col)[PDP_RESULT_FACTOR_NAN_MARKER])[0][0]
+        else:
+            idx = np.where(np.isnan(tmp[encoded_col]))[0][0]
+        # orig_null_value = tmp.from_num_to_factor(encoded_col)[tmp[encoded_col][idx]] if is_factor else tmp[encoded_col][idx]
+        orig_null_value = PDP_RESULT_FACTOR_NAN_MARKER if is_factor else np.nan
```

Comment:
In case that original value is NA, it prints a warning with it's value and informs that plotting NAs is not yet supported (will be in PUBDEV-8493) - that's what this block is about. In order to print the value of NA it's needed to find it in `tmp`, because NA values are calculated together with other ice plot points by using `include_na=True`. So `idx` is index of row that contains NA. If column is factor, I am looking for NA in form `PDP_RESULT_FACTOR_NAN_MARKER`, if it's numerical I am looking for NA in form `np.nan`. When I have `idx` I can extract respective y-axis value`tmp["mean_response"][idx]`.

In case it is not NA, it calculates the y-axis value only for that one point into `orig_tmp` and plots it, thats the else block.

I would leave there the commented line for debugging purposes, but can remove if it is annoying.